pkg/products/fuse/reconciler.go

```diff
@@ -244,56 +253,115 @@ func (r *Reconciler) reconcileViewFusePerms(ctx context.Context, client k8sclien
+
+	pgName := fmt.Sprintf("%s%s", constants.FusePostgresPrefix, rhmi.Name)
+	ns := rhmi.Namespace
+	postgres, err := croResources.ReconcilePostgres(ctx, client, defaultInstallationNamespace, rhmi.Spec.Type, "production", pgName, ns, pgName, ns, func(cr metav1.Object) error {
+		owner.AddIntegreatlyOwnerAnnotations(cr, rhmi)
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to reconcile postgres instance for fuse: %w", err)
+	}
+
+	// postgres provisioning is still in progress
+	if postgres.Status.Phase != types.PhaseComplete {
+		return integreatlyv1alpha1.PhaseAwaitingCloudResources, nil
+	}
+
+	// create the prometheus availability rule
+	if _, err = resources.CreatePostgresAvailabilityAlert(ctx, client, rhmi, postgres); err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create postgres prometheus alert for rhsso: %w", err)
+	}
+
+	// create the prometheus connectivity rule
+	if _, err = resources.CreatePostgresConnectivityAlert(ctx, client, rhmi, postgres); err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create postgres prometheus connectivity alert for rhsso : %s", err)
+	}
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
 // reconcileCustomResource ensures that the fuse custom resource exists
-func (r *Reconciler) reconcileCustomResource(ctx context.Context, installation *integreatlyv1alpha1.RHMI, client k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+func (r *Reconciler) reconcileCustomResource(ctx context.Context, rhmi *integreatlyv1alpha1.RHMI, client k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
 	r.logger.Info("Reconciling fuse custom resource")
 
-	cr := &syndesisv1alpha1.Syndesis{
+	pgName := fmt.Sprintf("%s%s", constants.FusePostgresPrefix, rhmi.Name)
+	// get the credential secret
+	postgresSec := &v1.Secret{}
+	if err := client.Get(ctx, k8sclient.ObjectKey{Name: pgName, Namespace: rhmi.Namespace}, postgresSec); err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to get postgres credential secret for fuse: %w", err)
+	}
+
+	// create the syndesis external database secret
+	synExternalDatabaseSec := &v1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
+			Name:      "syndesis-global-config",
 			Namespace: r.Config.GetNamespace(),
-			Name:      "integreatly",
 		},
 	}
+	_, err := controllerutil.CreateOrUpdate(ctx, client, synExternalDatabaseSec, func() error {
+		if synExternalDatabaseSec.Data == nil {
+			synExternalDatabaseSec.Data = map[string][]byte{}
+		}
+		synExternalDatabaseSec.Data["POSTGRESQL_PASSWORD"] = postgresSec.Data["password"]
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to reconcile fuse external database secret: %w", err)
+	}
 
-	intLimit := 0
+	cr := &syndesisv1beta1.Syndesis{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.Config.GetNamespace(),
+			Name:      "integreatly",
```

Comment:
Not sure about the cr name, any reason it has to be `"integreatly"` ?

Reply:
This was always the name of the Syndesis CR for RHMI. I didn't see any reason to change it.

Reply:
:+1: